recipes-connectivity/networkmanager/networkmanager-1.4.4/update-resolv-conf.patch

```diff
@@ -0,0 +1,176 @@
+-		update_resolv_conf (self, searches, nameservers, options, NULL, NM_DNS_MANAGER_RESOLV_CONF_MAN_UNMANAGED);
++		update_resolv_conf (self, RESOLV_CONF, searches, nameservers, options, NULL, NM_DNS_MANAGER_RESOLV_CONF_MAN_UNMANAGED);
+ 
+ 	/* signal that resolv.conf was changed */
+ 	if (update && result == SR_SUCCESS)
+@@ -1286,6 +1291,89 @@ forget_data (NMDnsManager *self, NMDnsIP
+ }
+ 
+ static gboolean
++xc_rewrite_resolv_conf (NMDnsManager *self,
++            const char *iface,
++            gpointer config,
++            gboolean no_caching,
++            GError **error)
++{
++       NMDnsManagerPrivate *priv;
++       NMResolvConfData rc;
++       GSList *iter, *vpn_configs = NULL, *dev_configs = NULL, *other_configs = NULL;
++       const char *nis_domain = NULL;
++       char **searches = NULL;
++       char **nameservers = NULL;
++       char **nis_servers = NULL;
++       char **options = NULL;
++       int num, i, len;
++       gboolean success = FALSE, caching = FALSE;
++       char resolv_path[PATH_MAX];
++
++       rc.nameservers = g_ptr_array_new ();
++       rc.searches = g_ptr_array_new ();
++       rc.options = g_ptr_array_new ();
++       rc.nis_domain = NULL;
++       rc.nis_servers = g_ptr_array_new ();
++
++       if (config){
++              if (NM_IS_IP4_CONFIG (config))
++                     merge_one_ip4_config (&rc, (NMIP4Config *) config);
++       }
++
++       /* Per 'man resolv.conf', the search list is limited to 6 domains
++        * totalling 256 characters.
++       */
++       num = MIN (rc.searches->len, 6);
++       for (i = 0, len = 0; i < num; i++) {
++              len += strlen (rc.searches->pdata[i]) + 1; /* +1 for spaces */
++              if (len > 256)
++                      break;
++       }
++       g_ptr_array_set_size (rc.searches, i);
++       if (rc.searches->len) {
++               g_ptr_array_add (rc.searches, NULL);
++               searches = (char **) g_ptr_array_free (rc.searches, FALSE);
++       } else
++              g_ptr_array_free (rc.searches, TRUE);
++
++       if (rc.nameservers->len) {
++               g_ptr_array_add (rc.nameservers, NULL);
++               nameservers = (char **) g_ptr_array_free (rc.nameservers, FALSE);
++       } else
++               g_ptr_array_free (rc.nameservers, TRUE);
++
++       if (rc.nis_servers->len) {
++               g_ptr_array_add (rc.nis_servers, NULL);
++               nis_servers = (char **) g_ptr_array_free (rc.nis_servers, FALSE);
++       } else
++               g_ptr_array_free (rc.nis_servers, TRUE);
++
++       if (rc.options->len) {
++               g_ptr_array_add (rc.options, NULL);
++               options = (char **) g_ptr_array_free (rc.options, FALSE);
++       } else
++               g_ptr_array_free (rc.options, TRUE);
++
++       nis_domain = rc.nis_domain;
++
++       strcpy(resolv_path, RESOLV_CONF);
++       strcat(resolv_path, ".");
++       strcat(resolv_path, iface);
++
++       if (success == FALSE)
```

Comment:
`success` is initialized to `FALSE` and is not used until it is tested here. No need to test.